### PR TITLE
Restrictions on mining

### PR DESCRIPTION
**Pull request by jonano614**
- restriction on count of shares from one connection per task
- restriction on count connection with the same address
- pool configuration is updated
- print miners function is updated

### CODE REVIEW

File: client/init.c [hunk 1]
```diff
@@ -220,12 +220,13 @@ void printUsage(char* appName)
 		"  -l             - output non zero balances of all accounts\n"
 		"  -m N           - use N CPU mining threads (default is 0)\n"
 		"  -p ip:port     - public address of this node\n"
-		"  -P ip:port:CFG - run the pool, bind to ip:port, CFG is miners:fee:reward:direct:maxip:fund\n"
+		"  -P ip:port:CFG - run the pool, bind to ip:port, CFG is miners:maxip:maxconn:fee:reward:direct:fund\n"
 		"                     miners - maximum allowed number of miners,\n"
```
Comment (xrdavies) [inline client/init.c]: It seams pool owners have to change the scripts to adapt such config. Need address it in change log.

File: client/pool.c [hunk 2]
```diff
@@ -508,6 +523,8 @@ static void register_new_miner(connection_list_element *connection, int index)
 		conn_data->miner = &new_miner->miner_data;
 		conn_data->state = ACTIVE_CONNECTION;
 	}
+
+	return 1;
```
Comment (xrdavies) [inline client/pool.c]: Usually use 0 for success, 1 for failure. Maybe keep the same rule it's better.

File: client/init.c [hunk 3]
```diff
@@ -220,12 +220,13 @@ void printUsage(char* appName)
 		"  -l             - output non zero balances of all accounts\n"
 		"  -m N           - use N CPU mining threads (default is 0)\n"
 		"  -p ip:port     - public address of this node\n"
-		"  -P ip:port:CFG - run the pool, bind to ip:port, CFG is miners:fee:reward:direct:maxip:fund\n"
+		"  -P ip:port:CFG - run the pool, bind to ip:port, CFG is miners:maxip:maxconn:fee:reward:direct:fund\n"
```
Comment (xrdavies) [inline client/init.c]: It seams pool owners have to change the scripts to adapt such config. Need address it in change log.

**Comment by xrdavies** [comment]
Thank you for your great efforts!

File: client/pool.c (hunk 2 above)
Comment (jonano614) [reply client/pool.c]: My logic was 1 - success, 0 - failure. And the check for result of function looks like:
if(!function())   //if(!success)
{.....}

**Pull request merged**